amr-wind/utilities/sampling/Sampling.cpp

```diff
@@ -220,6 +228,19 @@ void Sampling::write_netcdf()
     std::vector<size_t> start{nt, 0};
     std::vector<size_t> count{1, 0};
 
+    // Write the coordinates every time
+    std::vector<size_t> start3{nt, 0, 0};
+    std::vector<size_t> count3{1, 0, AMREX_SPACEDIM};
+    SamplerBase::SampleLocType locs;
+    for (const auto& obj : m_samplers) {
+        count[1] = obj->num_points();
+        auto grp = ncf.group(obj->label());
+        obj->sampling_locations(locs);
+        auto xyz = grp.var("coordinates");
+        count3[1] = obj->num_points();
+        xyz.put(&locs[0][0], start3, count3);
+    }
+
```

Comment:
This probably isn't doing what you intended. You still only have one set of locations in NetCDF file at any given time step. For stationary samplers, you're writing the same data over and over again every output timestep. Where as for non-stationary samplers, you'll end up overwriting previous locations with the latest positions and won't get a time series. 

What was probably intended is to allow LidarSampler to create additional time-series entries to allow output of the positions as a time series entry in NetCDF file.

Reply:
doesn't nt move the starting point in start 3? looks like line 236 should be deleted since 240 is what's used.

Reply:
Sorry, I overlooked the change in line 199. 

This is ugly though.

Reply:
@sayerhs , what would be a better way of doing this?

Reply:
@tonyinme I was hoping for a more flexible way of doing this so that stationary samplers didn't have to dump the same data over and over again, and providing more flexibility to specific samplers (like Lidar) to provide additional output. 

Using the example of `PlaneSampler` we currently have two functions to customize output behavior: 

https://github.com/Exawind/amr-wind/blob/d67e6e2db9b4d73acc05d516ba23d5ab0f6b839f/amr-wind/utilities/sampling/PlaneSampler.cpp#L75
https://github.com/Exawind/amr-wind/blob/d67e6e2db9b4d73acc05d516ba23d5ab0f6b839f/amr-wind/utilities/sampling/PlaneSampler.cpp#L88

My proposal would be to add these method overrides to `LidarSampler` customize the data output. Add 

- In addition to start/stop, also add the tables of azimuth and elevation
- Delcare a new entry `position` that has the exact coordinates of sampling as a function of time. This is probably not needed as it can be recomputed from the azimuth/elevation tables, but is nice to output. 
- use `populate_netcdf_metadata` to output the tables once for the LidarSampler
- Create a new virtual method (say `output_netcdf_data`) - You will have to add this to SamplerBase also - and then customize this to output the sampling locations for the LidarSampler. Call this in `Sampling.cpp` after the field data is output. 

This retains the original behavior for stationary samplers, but also provides a flexible way to customize output behavior for lidars and other samplers in future.

Reply:
Thanks for the suggested workflow @sayerhs. This has been incorporated to the latest version.